src/app/core/ui/search/search.component.ts

```diff
@@ -28,9 +37,91 @@ export class SearchComponent implements OnInit {
 
   ngOnInit() {
     this.createSearchIndex();
+
+    this.searchStringChanged
+      .pipe(debounceTime(this.INPUT_DEBOUNCE_TIME_MS))
+      .subscribe((searchString) => {
+        this.searchQuery.next(this.startSearch(searchString));
+      });
+
+    this.searchQuery
+      .pipe(
+        switchMap((value) => {
+          return value;
+        })
+      )
+      .subscribe(
+        (result: { queryResults: any; searchTerms: string[] }) => {
+          this.handleSearchQueryResult(result);
+        },
+        (error) => {
+          console.error("SearchQuery failed", error);
+        }
+      );
+  }
+
+  async startSearch(searchString: string): Promise<any> {
+    this.isSearching = true;
+    this.results = [];
+
+    if (!searchString) {
+      return Promise.resolve();
+    }
+
+    searchString = searchString.toLowerCase();
+
+    if (!this.isRelevantSearchInput(searchString)) {
+      return Promise.resolve();
+    }
+
+    const searchTerms = searchString.split(" ");
+
+    const queryResults = await this.db.query("search_index/by_name", {
+      startkey: searchTerms[0],
+      endkey: searchTerms[0] + "\ufff0",
+      include_docs: true,
+    });
+
+    return {
+      queryResults,
+      searchTerms,
+    };
   }
 
-  private createSearchIndex() {
+  clickOption(optionElement) {
+    // simulate a click on the EntityBlock inside the selected option element
+    optionElement._element.nativeElement.children["0"].children["0"].click();
+    if (this.showSearchToolbar === true) {
+      this.showSearchToolbar = false;
+    }
+  }
+
+  toggleSearchToolbar() {
+    this.showSearchToolbar = !this.showSearchToolbar;
+  }
+
+  searchInputChanged(searchString) {
+    this.searchStringChanged.next(searchString);
```

Comment:
can we call this directly in the template (and reduce the number of methods in the class here)?

Reply:
done.